src/attack.rs

```diff
@@ -256,16 +265,37 @@ fn player_flop(
             if input.just_pressed(PlayerAction::FlopAttack) {
                 state.set(State::Attacking);
 
+                let attack_entity = commands
+                    .spawn_bundle(TransformBundle::default())
+                    .insert(Sensor(true))
+                    .insert(ActiveEvents::COLLISION_EVENTS)
+                    .insert(ActiveCollisionTypes::default() | ActiveCollisionTypes::STATIC_STATIC)
+                    .insert(CollisionGroups::new(
+                        BodyLayers::PLAYER_ATTACK,
+                        BodyLayers::ENEMY,
+                    ))
+                    .insert(Attack { damage: 10 })
+                    .insert(AttackFrames {
+                        startup: 0,
+                        active: 3,
+                        recovery: 4,
+                    })
+                    .id();
+                commands.entity(entity).push_children(&[attack_entity]);
+                //TODO: define hitbox size and placement through resources
+
+                //maybe move audio effects?
```

Comment:
Possibly! It depends on what you have in mind :smile: 

Maybe adding a system that matches the current on-screen animations? I haven't checked the system, in this case, they'd need to be connected to a state.

Reply:
Not sure I have anything in mind yet, just noting that it feels a bit awkward that it lives here, I think this might make sense to either trigger based off attack actionlike state or when we split out the movement intent from attacks also use a similar strategy to split out the sound triggering intent of attacks.

Reply:
Difficult to say :thinking: The reason is that this is already a decoupled logic - `FighterStateEffectsPlayback` does not play any effect itself, but it adds a component, in turn this component is picked up by a separate system (`fighter_sound_effect`).

Having said that, I definitely find ugly the surrounding conditionals. They can be offloaded to the `fighter_sound_effect` system, by passing the `Handle<FighterMeta>`, so that the audio system can find out the asset by itself. On the other hand, this design leads to one system is saying to another "here's the metadata; maybe you'll play a sound, maybe you won't". This can be fine, it's a matter of taste.

Let me know if you have any preference, since I think all the approaches are equally valid.